app/src/main/java/ml/docilealligator/infinityforreddit/network/NetworkModule.java

```diff
@@ -0,0 +1,215 @@
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("imgur")
+    @Singleton
+    Retrofit provideImgurRetrofit() {
+        return new Retrofit.Builder()
+                .baseUrl(APIUtils.IMGUR_API_BASE_URI)
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("pushshift")
+    @Singleton
+    Retrofit providePushshiftRetrofit() {
+        return new Retrofit.Builder()
+                .baseUrl(APIUtils.PUSHSHIFT_API_BASE_URI)
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("reveddit")
+    @Singleton
+    Retrofit provideRevedditRetrofit() {
+        return new Retrofit.Builder()
+                .baseUrl(APIUtils.REVEDDIT_API_BASE_URI)
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("vReddIt")
+    @Singleton
+    Retrofit provideVReddItRetrofit() {
+        return new Retrofit.Builder()
+                .baseUrl("http://localhost/")
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("strapi")
+    @Singleton
+    Retrofit providestrapiRetrofit(@Named("default") OkHttpClient okHttpClient) {
+        return new Retrofit.Builder()
+                .baseUrl(APIUtils.STRAPI_BASE_URI)
+                .client(okHttpClient)
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Named("streamable")
+    @Singleton
+    Retrofit provideStreamableRetrofit(@Named("default") OkHttpClient okHttpClient) {
+        return new Retrofit.Builder()
+                .baseUrl(APIUtils.STREAMABLE_API_BASE_URI)
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    StreamableAPI provideStreamableApi(@Named("streamable") Retrofit streamableRetrofit) {
+        return streamableRetrofit.create(StreamableAPI.class);
+    }
+
+    @Provides
+    @Named("default")
+    @Singleton
+    OkHttpClient provideOkHttpClient(@Named("no_oauth") Retrofit retrofit, RedditDataRoomDatabase accountRoomDatabase,
+                                     @Named("current_account") SharedPreferences currentAccountSharedPreferences) {
+        OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
+        okHttpClientBuilder.authenticator(new AccessTokenAuthenticator(retrofit, accountRoomDatabase, currentAccountSharedPreferences))
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .writeTimeout(30, TimeUnit.SECONDS)
+                .connectionPool(new ConnectionPool(0, 1, TimeUnit.NANOSECONDS));
```

Comment:
Not related to this pr, but are you sure this is necessary? Usually people try to reuse connections as much a possible but this completely prevents any reuse.

Reply:
Yes. It's necessary. Prior to adding this, many users reported that Infinity could not connect to the Reddit server after some time.